bosh-director/lib/bosh/director/rendered_job_templates_cleaner.rb

```diff
@@ -1,20 +1,33 @@
 module Bosh::Director
   class RenderedJobTemplatesCleaner
-    def initialize(instance_model, blobstore)
+    def initialize(instance_model, blobstore, logger)
       @instance_model = instance_model
       @blobstore = blobstore
+      @logger = logger
     end
 
     def clean
       @instance_model.stale_rendered_templates_archives.each do |archive|
-        @blobstore.delete(archive.blobstore_id)
+        if @blobstore.exists? archive.blobstore_id
```

Comment:
@dk, yes, but one thing is we need to change all kinds of blobstore clients(dav, local, s3) to raise NotFound error instead of BlobstoreError, this may introduce more changes, if you think is ok, we will change the Blobstore client then